### PR TITLE
Added picocli, -d argument and added the main class to jar's manifest

### PR DESCRIPTION
**Pull request by hbelmiro**
This PR:
* Removes the .idea directory
* Makes the application runnable through `mvn spring-boot:run` or `java -jar`.
* Adds support to picocli to parse command line options
* Adds the `-d`/`--destination` option so the user can choose the destination of the generated microservice
* Updates README.md with the above information

### CODE REVIEW

**Comment by Vipin-Sharma** [comment]
Thanks @hbelmiro this is the first PR to MSG!
Allow me some time to review this.

File: README.adoc [hunk 1]
```diff
@@ -23,16 +23,52 @@ Sometimes we need to hardcode some values in SQL, for that we can use syntax lik
...
-          - Linux: /home/[username]
-          - Mac: /Users/[username]
+There are two ways you can run MSG:
+
+. From sources with Maven:
+
++
+[source,shell]
+----
+mvn spring-boot:run -Dspring-boot.run.arguments="--destination /tmp/test"
```
Comment (Vipin-Sharma) [inline README.adoc:46]: This works fine, just one minor improvement I would suggest, in case user does not specify destination we can generate code in user home and at the end of execution we display where is code generated. This way it should be only `mvn spring-boot:run` for user, not confusion/complexity at all. Destination works well for advance users.

File: README.adoc [hunk 2]
```diff
@@ -23,16 +23,52 @@ Sometimes we need to hardcode some values in SQL, for that we can use syntax lik
...
+
+. Running the jar:
+
++
+[source,shell]
+----
+java -jar MSG-1.0-SNAPSHOT.jar --destination /tmp/test
+----
+
+Run MSG to generate the microservice application. The microservice application will be generated in the user home directory.
```
Comment (Vipin-Sharma) [inline README.adoc:57]: We need to reword this as in case destination is given it may not be user home directory.

File: src/main/java/com/jfeatures/msg/sql/ReadFileFromResources.java [hunk 3]
```diff
@@ -101,17 +96,11 @@ public static Properties readPropertiesFromFile(String propertyFileName) throws
...
-        Path path = Path.of(resource.toURI());
-        String fileContent = "";
-        try {
-            fileContent = new String(Files.readAllBytes(path));
+    public static String readFileFromResources(String fileName) {
+        try (var inputStream = ReadFileFromResources.class.getClassLoader().getResourceAsStream(fileName)) {
+            return new String(inputStream.readAllBytes());
         } catch (IOException e) {
-            log.severe("Error reading file from resources" + e.getMessage());
+            throw new UncheckedIOException(e);
```
Comment (Vipin-Sharma) [inline src/main/java/com/jfeatures/msg/sql/ReadFileFromResources.java:103]: This is good addition, isn't it good idea to keep a logger statement as well? that gives a nice English language message about error.

**Review by Vipin-Sharma — CHANGES_REQUESTED** [review]

File: README.adoc (hunk 1 above)
Comment (Vipin-Sharma) [reply README.adoc:46]: I see it works well with men spring-boot, exactly what I was expecting my comment above, but why are we using spring-boot here? I think here we are using spring boot just for packaging and running code, right?

Comment (hbelmiro) [reply README.adoc:46]: Yes, spring-boot:run will be only used by the developer of MSG. The user will normally run with java -jar.

File: README.adoc (hunk 2 above)
Comment (hbelmiro) [reply README.adoc:57]: Done.

File: src/main/java/com/jfeatures/msg/sql/ReadFileFromResources.java (hunk 3 above)
Comment (hbelmiro) [reply src/main/java/com/jfeatures/msg/sql/ReadFileFromResources.java]: What do you think of the following?

```java
throw new UncheckedIOException("Error reading file from resources: " + e.getMessage(), e);
```

This way we log only one friendly message that includes the cause.
I pushed that. If you think it's better leaving the severe message, just let me know.

Comment (hbelmiro) [reply src/main/java/com/jfeatures/msg/sql/ReadFileFromResources.java]: What do you think of the following?

```java
throw new UncheckedIOException("Error reading file from resources: " + e.getMessage(), e);
```

This way we log only one friendly message that includes the cause.
I pushed that. If you think it's better leaving the severe message, just let me know.

File: README.adoc (hunk 1 above)
Comment (Vipin-Sharma) [reply README.adoc:46]: I see what you are saying, I think this is good for now.
MSG has spring-boot dependency because we need to use spring boot annotations while generating the code. MSG is not a spring boot application, it is a standalone JVM application. 
`spring-boot:run` is typically used to start a spring boot microservice, while we are using it to run a standalone JVM application, I am not sure if it is a good idea or not, it needs more thinking but for now we can use this. Created issue https://github.com/Vipin-Sharma/MSG/issues/21 for this.

**Review by Vipin-Sharma — APPROVED** [review]
